src/client_impl.cc

```diff
@@ -53,7 +35,9 @@ MixerClientImpl::MixerClientImpl(const MixerClientOptions &options)
   quota_cache_ =
       std::unique_ptr<QuotaCache>(new QuotaCache(options.quota_options));
 
-  deduplication_id_base_ = GenerateUUID();
+  if (options_.uuid_generate_func) {
```

Comment:
we should provide a default uuid function.

For Linux,  we can just try to read "/proc/" as Envoy.